### ISSUE TITLE
[NcActions] doesn't fully support changing type during open state

### ISSUE THREAD

**Issue by ShGKme** [nextcloud-libraries/nextcloud-vue#5334]
`NcActions` can have a different type: navigation (expanded list), application menu, form dialog.
It not only changes attributes for a11y but also keyboard navigation and `focus-trap` integration.

But it doesn't work well when it is changed during menu usage.

Example — renaming an address book in Contacts app.
1. Initially it is a menu without a focus trap. Opening this menu pauses the global focus trap (e.g. current modal). It is needed because in a DOM menu is rendered outside the modal with a focus trap, and otherwise modal's trap breaks navigation in actions. It is supposed to be unpaused on close.
   ![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/25978914/1879dc80-5287-4444-8c66-cf06214a8cce)
2. On "Edit" click it, this menu item is replaced with `input`. With a form, this popup is no longer a menu but a dialog form. Dialog is supposed to have its own focus trap. But it doesn't, because a focus trap should have been enabled initially in `NcPopover`.
  ![image](https://github.com/nextcloud-libraries/nextcloud-vue/assets/25978914/d63f74dd-21b1-4166-850e-b202c48c6c18)
3. When the dialog is closed, the global focus trap is not unpaused. Because it is not needed for the dialog. So the focus trap of modal is broken now.

A solution could be:
1. Add `focusTrap` setting change support to `NcPopover` so we can enable `focusTrap` for existed `dialog`
2. Always unpause the focus trap in `NcActions` on close if it was paused before, not matter if was expected.

See also: https://github.com/nextcloud/contacts/issues/3812
